src/app/reactUtils.ts

```diff
@@ -130,3 +132,62 @@ export const onEscHandler = (element: React.RefObject<HTMLElement>, isEnabled: b
 export const useOnEsc = (element: React.RefObject<HTMLElement>, isEnabled: boolean, cb: () => void) => {
   React.useEffect(onEscHandler(element, isEnabled, cb), [element, isEnabled]);
 };
+
+export const useMatchMobileQuery = () => {
+  const matchMedia = assertWindow().matchMedia(theme.breakpoints.mobileQuery);
+  const [isMobile, setIsMobile] = React.useState(matchMedia.matches);
+
+  const listener = React.useCallback((e: EventListener) => {
+    if ((e as MediaQueryListEvent).matches) {
+      setIsMobile(true);
+    } else {
+      setIsMobile(false);
+    }
+  }, []);
+
+  React.useEffect(() => {
+    matchMedia.addListener(listener);
+    return () => { matchMedia.removeListener(listener); };
+  }, [listener, matchMedia]);
+
+  return isMobile;
+};
+
+export const useDebouncedWindowSize = () => {
+  const window = assertWindow();
+  const timeout = React.useRef(0);
+  const [size, setSize] = React.useState([window.innerWidth, window.innerHeight]);
+
+  React.useLayoutEffect(() => {
+    const updateSize = () => {
+      clearTimeout(timeout.current);
+      timeout.current = setTimeout(() => {
+        setSize([window.innerWidth, window.innerHeight]);
+      }, 50);
+    };
+    window.addEventListener('resize', updateSize);
+    return () => {
+      clearTimeout(timeout.current);
+      window.removeEventListener('resize', updateSize);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return size;
+};
+
+export const useOnScrollTopOffset = () => {
+  const document = assertDocument();
+  const [topOffset, setTopOffset] = React.useState(0);
+
+  const listener = React.useCallback(() => {
+    setTopOffset(document.scrollingElement ? document.scrollingElement.scrollTop : 0);
+  }, [document]);
+
+  React.useEffect(() => {
+    document.addEventListener('scroll', listener);
+    return () => { document.removeEventListener('scroll', listener); };
+  }, [document, listener]);
```

Comment:
Because nudge component is always rendered (just returns null sometimes), this will attach a permanent scroll listener to the document which is definitely not good

Reply:
I feel like the nudge component should be rendered conditionally to avoid any unnecessary dom interactions.

Reply:
I'm not sure if this is "definitely not good", but it is easy to avoid so I'll render it conditionally

Reply:
That's true. I overexaggerated, sorry.